README.md

```diff
@@ -301,14 +302,30 @@ UI_BAKERY_PORT=80
 
 1. Set variable `UI_BAKERY_SAML_ENABLED=true`
 
-1. You can add a role mapping from identity provider role to UI Bakery role via env variable:
+1. You can set the variable `UI_BAKERY_SAML_LOGIN_AUTO` to true to enable automatic login. Any unauthorized user will be redirected to SAML login flow.
+
+### SAML roles synchronization
+
+By default, UI Bakery will not sync any roles provided by the Identity Provider. 
 
+1. To enable roles synchronization, set the variable `UI_BAKERY_SAML_SYNC_ROLES=true`.
+1. By default, UI Bakery will try to match roles by names. If a role match is found, the matched role will be assigned to the user.
+   - Additionally, you can configure a role mapping from identity provider role id/name to a UI Bakery role:
+
+      ```bash
+      UI_BAKERY_ROLE_MAPPING=identityRoleName->bakeryRoleName,identityRoleName2->bakeryRoleName2
+      ```
+1. By default, UI Bakery will not remove roles not provided by the Identity Provider. If your setup requires a complete syncronization, when UI Bakery reassigns all roles, removing existing ones and adding new ones received from Identity Provider, use the following variable:
    ```bash
-   UI_BAKERY_ROLE_MAPPING=identityRoleName->bakeryRoleName,identityRoleName2->bakeryRoleName2
+    UI_BAKERY_SAML_HARD_SYNC_ROLES=true
```

Comment:
let's add message that without this setting the "user" will be applied to user if no role match

Reply:
This is not exactly the behaviour. Roles will be left as is if no roles provided by the identity server.

Reply:
added a note